fed/_private/queue.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2023 The RayFed Team
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import threading
+from collections import deque
+import time
+import logging
+
+
+logger = logging.getLogger(__name__)
+
+STOP_SYMBOL = False
+
+
+class MessageQueue:
+    def __init__(self, msg_handler, failure_handler=None, name=''):
+        assert callable(msg_handler), "msg_handler must be a callable function"
+
+        self._queue = deque()
+        self._msg_handler = msg_handler
+        self._failure_handler = failure_handler
+        self._thread = None
+        self._name = name
```

Comment:
What this name indicate?

Reply:
As this util class is too general from the class name. It's better to add the detail comments on the fields.

Reply:
It's the sub-thread's name. I'll re-name and add explanation.